astropy/coordinates/angles.py

```diff
@@ -350,11 +350,17 @@ def plain_unit_format(val):
                 "The unit value provided is not an angular unit.")
 
         def do_format(val):
-            s = func(float(val))
-            if alwayssign and not s.startswith('-'):
-                s = '+' + s
-            if format == 'latex':
-                s = f'${s}$'
+            #Check if value is finite to avoid ValueErrors when turning it into
+            #a hexagesimal string.
+            if np.isfinite(val):
+                s = func(float(val))
+                if alwayssign and not s.startswith('-'):
+                    s = '+' + s
+                if format == 'latex':
+                    s = f'${s}$'
+                return s
+            unit_string = unit.to_string(format=format)
+            s = f"{val} {unit_string}"
```

Comment:
Just a note: ``unit_string`` is never reused, so we can avoid assigning a variable.

```suggestion
            s = f"{val} {unit.to_string(format=format)}"
```